src/proto/smb.rs

```diff
@@ -0,0 +1,1143 @@
+            }
+            // 0x73 => {
+            //     // Setup
+            //     SMB1Payload::SetupRequest(SMB2SetupRequest::new())
+            // }
+            _ => None?,
+        });
+        self.payload.as_mut()
+    }
+}
+
+#[derive(Debug, Clone, PartialEq)]
+struct SMB1Dialect {
+    buffer_format: u8,
+    dialect_string: String,
+}
+
+#[derive(Debug, Clone, Copy)]
+enum SMB1NegotiateRequestState {
+    WordCount,
+    ByteCount,
+    Dialects,
+    End,
+}
+
+#[derive(Debug, Clone)]
+struct SMB1NegotiateRequest {
+    // DISSECTION
+    d: PacketDissector<SMB1NegotiateRequestState>,
+    _tmp_dialect: Option<SMB1Dialect>,
+    // STRUCT
+    word_count: u8,
+    byte_count: u16,
+    dialects: Vec<SMB1Dialect>,
+}
+
+impl Packet for SMB1NegotiateRequest {
+    fn new() -> SMB1NegotiateRequest {
+        Self {
+            d: PacketDissector::new(SMB1NegotiateRequestState::WordCount),
+            _tmp_dialect: None,
+            word_count: 0,
+            byte_count: 0,
+            dialects: Vec::new(),
+        }
+    }
+
+    fn parse(&mut self, byte: &u8) {
+        match self.d.state {
+            SMB1NegotiateRequestState::WordCount => {
+                self.word_count = *byte;
+                self.d.next_state(SMB1NegotiateRequestState::ByteCount);
+            }
+            SMB1NegotiateRequestState::ByteCount => {
+                self.byte_count =
+                    self.d
+                        .read_ule16(byte, self.byte_count, SMB1NegotiateRequestState::Dialects);
+            }
+            SMB1NegotiateRequestState::Dialects => {
+                self.d.i += 1;
+                match self._tmp_dialect.as_mut() {
+                    Some(dial) => {
+                        if *byte == 0 {
+                            // Final nul byte: dialect is finished
+                            self.dialects.push(dial.clone());
+                            self._tmp_dialect = None;
+                            self.d.next_state_when_i_reaches(
+                                SMB1NegotiateRequestState::End,
+                                self.byte_count as usize,
+                            );
+                        } else {
+                            dial.dialect_string.push(*byte as char);
+                        }
+                    }
+                    None => {
+                        self._tmp_dialect = Some(SMB1Dialect {
+                            buffer_format: *byte,
+                            dialect_string: String::new(),
+                        });
+                    }
+                }
+            }
+            SMB1NegotiateRequestState::End => {}
+        }
+    }
+
+    fn repl(&self) -> Option<Vec<u8>> {
+        if !matches!(self.d.state, SMB1NegotiateRequestState::End) {
+            return None;
+        }
+        let mut resp: Vec<u8> = Vec::new();
+        let time: u64 = (EPOCH_1601 + chrono::Utc::now().timestamp() as u64) * (1e7 as u64);
```

Comment:
You can use `SystemTime::now().duration_since(SystemTime::UNIX_EPOCH)` (see https://doc.rust-lang.org/std/time/struct.SystemTime.html).

Reply:
Will do.
I had found chrono in Cargo.toml so apparently this is used somewhere :P

I looked into using something like `duration_since(...1601 EPOCH...)` but the format [looks kinda crazy](https://github.com/rust-lang/rust/blob/9a60099cc43c8a07abb280be323d1ed9afc27f2c/library/std/src/sys/windows/time.rs#L27)